pkgs/development/androidndk-pkgs/androidndk-pkgs.nix

```diff
@@ -48,35 +48,52 @@ let
   hostInfo = ndkInfoFun stdenv.hostPlatform;
   targetInfo = ndkInfoFun stdenv.targetPlatform;
 
-  prefix = lib.optionalString (stdenv.targetPlatform != stdenv.hostPlatform) (stdenv.targetPlatform.config + "-");
+  sdkVer = stdenv.targetPlatform.sdkVer;
+  prefix = lib.optionalString (stdenv.targetPlatform != stdenv.hostPlatform) (stdenv.targetPlatform.config);
 in
 
 rec {
   # Misc tools
-  binaries = runCommand "ndk-toolchain-binutils" {
-    pname = "ndk-toolchain-binutils";
+  binaries = stdenv.mkDerivation {
+    pname = "ndk-toolchain";
     inherit (androidndk) version;
     isClang = true; # clang based cc, but bintools ld
-    nativeBuildInputs = [ makeWrapper ];
+    nativeBuildInputs = [ makeWrapper python autoPatchelfHook ];
+
     propagatedBuildInputs = [ androidndk ];
-  } ''
-    mkdir -p $out/bin
+    dontUnpack = true;
+    dontBuild = true;
+    dontStrip = true;
+    dontConfigure = true;
+    dontPatch = true;
+    autoPatchelfIgnoreMissingDeps = true;
+    installPhase = ''
+      ${androidndk}/libexec/android-sdk/ndk-bundle/build/tools/make-standalone-toolchain.sh --arch=${targetInfo.arch} --install-dir=$out/toolchain --platform=${sdkVer} --force
+      ln -vfs $out/toolchain/sysroot/usr/lib $out/lib
+      ln -s $out/toolchain/sysroot/usr/lib/${targetInfo.triple}/*.so $out/lib/
+      ln -s $out/toolchain/sysroot/usr/lib/${targetInfo.triple}/*.a $out/lib/
+      chmod +w $out/lib/*
+      ln -s $out/toolchain/sysroot/usr/lib/${targetInfo.triple}/${sdkVer}/*.so $out/lib/
+      ln -s $out/toolchain/sysroot/usr/lib/${targetInfo.triple}/${sdkVer}/*.o $out/lib/
```

Comment:
because this is the `binaries` not `libraries`, we shouldn't have target stuff in them. For LLVM-based toolchains (and someday also GCC) we should be able to cleanly separate the two.

Reply:
So what should I do? Move this stuff to libraries?

Reply:
Yeah, Maybe run the toolchain script in both and only install certain things in one, and other things in the other.